homeassistant/components/ezviz/camera.py

```diff
@@ -0,0 +1,276 @@
+                camera_password = conf_cameras[camera_serial]["password"]
+                camera_rtsp_stream = "rtsp://{}:{}@{}:{}".format(
+                    camera_username,
+                    camera_password,
+                    camera["localIp"],
+                    camera["localRtspPort"],
+                )
+                _LOGGER.debug(
+                    "Camera %s source stream: %s", camera["serial"], camera_rtsp_stream
+                )
+
+            else:
+                _LOGGER.error(
+                    "I found a camera (%s) but it is not configured. Please configure it if you wish to see the appropriate stream. Conf cameras: %s",
+                    camera_serial,
+                    conf_cameras,
+                )
+
+            async_add_entities(
+                [
+                    EzvizCamera(
+                        hass,
+                        camera_username,
+                        camera_password,
+                        camera_rtsp_stream,
+                        camera["serial"],
+                        camera["name"],
+                        camera["deviceType"],
+                        camera["version"],
+                        camera["status"],
+                        camera["createTime"],
+                        camera["category"],
+                        camera["subCategory"],
+                        camera["customType"],
+                        camera["localIp"],
+                        camera["netIp"],
+                        camera["localRtspPort"],
+                        camera["wanIp"],
+                        camera["netType"],
+                    )
+                ]
+            )
+    except PyEzvizError as exp:
+        _LOGGER.error(exp)
+        return
+
+
+class EzvizCamera(Camera):
+    """An implementation of a Foscam IP camera."""
+
+    def __init__(
+        self,
+        hass,
+        username,
+        password,
+        rtsp_stream,
+        serial,
+        name,
+        device_type,
+        version,
+        status,
+        create_time,
+        category,
+        sub_category,
+        custom_type,
+        local_ip,
+        net_ip,
+        local_rtsp_port,
+        wan_ip,
+        net_type,
+    ):
+        """Initialize an Ezviz camera."""
+        super().__init__()
+
+        self._username = username
+        self._password = password
+        self._rtsp_stream = rtsp_stream
+
+        self._serial = serial
+        self._name = name
+        self._type = device_type
+        self._version = version
+        self._status = status
+        self._create_time = create_time
+        self._category = category
+        self._sub_category = sub_category
+        self._custom_type = custom_type
+        self._local_ip = local_ip
+        self._net_ip = net_ip
+        self._local_rtsp_port = local_rtsp_port
+        self._wan_ip = wan_ip
+        self._net_type = net_type
+
+        self._ffmpeg = hass.data[DATA_FFMPEG]
```

Comment:
Implement `async def async_added_to_hass(self):` and move this assignment there.

Reply:
just the _ffmpeg one ?
As I read in the doc, async_added_to_hass has to be called when an entity has their entity_id and hass object assigned, before it is written to the state machine for the first time. 

I don't really get what it will do tbh

Reply:
You don't have access to the `hass` instance at this point, so you'll have to move assignment of this variable to when it is available (which is populated before `async_added_to_hass` is called) or get it somewhere else and pass it directly to `init`.

Reply:
Ok so I initialised it in __init__, and then set it in added_to_hass

I tried to read other camera.py code, they don't seem to implement it this way, but it works :)

Reply:
We can pass the ffmpg data as an argument to the entity.